pkg/controller/operands/ssp_test.go

```diff
@@ -70,822 +79,549 @@ var _ = Describe("SSP Operands", func() {
+						foundResource),
+				).To(BeNil())
+
+				Expect(existingResource.Spec.TemplateValidator.Placement.Affinity.NodeAffinity).ToNot(BeZero())
+				Expect(existingResource.Spec.TemplateValidator.Placement.Tolerations).To(HaveLen(2))
+				Expect(existingResource.Spec.TemplateValidator.Placement.NodeSelector["key1"]).Should(Equal("value1"))
+
+				Expect(foundResource.Spec.TemplateValidator.Placement.Affinity.NodeAffinity).ToNot(BeNil())
+				Expect(foundResource.Spec.TemplateValidator.Placement.Tolerations).To(HaveLen(3))
+				Expect(foundResource.Spec.TemplateValidator.Placement.NodeSelector["key1"]).Should(Equal("something else"))
+
+				Expect(req.Conditions).To(BeEmpty())
+			})
 
-			foundResource := &sspv1.KubevirtNodeLabellerBundle{}
-			Expect(
-				cl.Get(context.TODO(),
-					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
-					foundResource),
-			).To(BeNil())
+			It("should overwrite node placement if directly set on SSP CR", func() {
+				hco.Spec.Infra = hcov1beta1.HyperConvergedConfig{NodePlacement: commonTestUtils.NewHyperConvergedConfig()}
+				existingResource := NewSSP(hco, commonTestUtils.Namespace)
+
+				// mock a reconciliation triggered by a change in NewKubeVirtNodeLabellerBundle CR
+				req.HCOTriggered = false
+
+				// now, modify NodeLabeller node placement
+				seconds3 := int64(3)
+				existingResource.Spec.TemplateValidator.Placement.Tolerations = append(hco.Spec.Infra.NodePlacement.Tolerations, corev1.Toleration{
+					Key: "key3", Operator: "operator3", Value: "value3", Effect: "effect3", TolerationSeconds: &seconds3,
+				})
+				existingResource.Spec.TemplateValidator.Placement.NodeSelector["key1"] = "BADvalue1"
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := newSspHandler(cl, commonTestUtils.GetScheme())
+				res := handler.ensure(req)
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeTrue())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &sspv1beta1.SSP{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
+
+				Expect(existingResource.Spec.TemplateValidator.Placement.Tolerations).To(HaveLen(3))
+				Expect(existingResource.Spec.TemplateValidator.Placement.NodeSelector["key1"]).Should(Equal("BADvalue1"))
+
+				Expect(foundResource.Spec.TemplateValidator.Placement.Tolerations).To(HaveLen(2))
+				Expect(foundResource.Spec.TemplateValidator.Placement.NodeSelector["key1"]).Should(Equal("value1"))
+
+				Expect(req.Conditions).To(BeEmpty())
+			})
+		})
 
-			Expect(existingResource.Spec.Affinity.NodeAffinity).ToNot(BeNil())
-			Expect(existingResource.Spec.Tolerations).To(HaveLen(2))
-			Expect(existingResource.Spec.NodeSelector["key1"]).Should(Equal("value1"))
+		Context("SSP Upgrade", func() {
```

Comment:
Maybe it worth to add a test case(s) where the SSP CR already exist (for future upgrades). Although this is a very unlikely to be relevant.

Reply:
Not sure what is the test case to be added? Meaning, what logic to we want to test?

Reply:
Just make sure that the HyperConverged CR 's node placement related data is copied to the SSP CR

Reply:
This behavior is already covered in various tests under the "Node placement" context.